gremlin-go/driver/graphBinary.go

```diff
@@ -1044,6 +1078,18 @@ func timeReader(data *[]byte, i *int) (interface{}, error) {
 	return time.UnixMilli(readLongSafe(data, i)), nil
 }
 
+func offsetDateTimeReader(data *[]byte, i *int) (interface{}, error) {
+	year := readIntSafe(data, i)
+	month := readByteSafe(data, i)
+	day := readByteSafe(data, i)
+	ns := readLongSafe(data, i)
+	offset := readIntSafe(data, i)
+	// only way to pass offset info, timezone display is fixed to UTC as consequence (offset is calculated properly)
+	loc := time.FixedZone("UTC", int(offset))
+	datetime := time.Date(int(year), time.Month(month), int(day), 0, 0, 0, int(ns), loc)
```

Comment:
int(ns) may be an issue for times which are late in the day. According to the io docs, the "time" value can be from 0 to 86,399,999,999,999. The max size of `int` is only guaranteed to be 2,147,483,647, although on most 64-bit machines it will be a 64 bit integer in practice.

Reply:
Unfortunately the Date constructor requires `int`, it's more like a caveat of go, not sure if it's avoidable.

Reply:
I think we need to break the 64bit `ns` into `hours`, `minutes`, `seconds`, and `nanos` to ensure none of them will overflow on systems with 32bit ints. We can't allow those datetimes to overload on deserialization.